### PR TITLE
feat: added Github deploy action

### LINKED ISSUE #71: deploy svelte generated website

### PR DESCRIPTION
**Pull request by RemcoHalman**
Closes #71 

Added the `deploy.yml`. 

Could you check if it will deploy correctly `line 25`. Or if I need to change this to be the `src` folder.

### CODE REVIEW

File: .github/workflows/deploy.yml
Comment (eddiejaoude) [inline .github/workflows/deploy.yml:5]: This should be `main` branch

File: .github/workflows/deploy.yml
Comment (eddiejaoude) [inline .github/workflows/deploy.yml:25]: I think this should be `public`

File: .github/workflows/deploy.yml
Comment (eddiejaoude) [inline .github/workflows/deploy.yml:24]: This should be `gh-pages`

**Comment by eddiejaoude** [comment]
Thanks @RemcoHalman 👍 - I left a few small comments for some tweaks.

**Comment by RemcoHalman** [comment]
@eddiejaoude Updated with the tweaks

File: .github/workflows/deploy.yml
Comment (BOLT04) [reply .github/workflows/deploy.yml]: I think so too, the `index.html` and other resources like images need to be deployed too 😃

**Comment by BOLT04** [comment]
@RemcoHalman thank you for this contribution 👍. I'm not sure why cypress isn't running for your PR, but everything seems fine so I think we can merge this 😃.

@eddiejaoude what do you think?

**Review by BOLT04 — APPROVED** [review]

**Comment by eddiejaoude** [comment]
Thanks @RemcoHalman .

Interesting spot @BOLT04 , cypress should run when PRs are `opened` (we might need to add more events to this as PRs are updated and cypress should be re-run), this is the config we use

```yaml
on:
  push:
  pull_request:
    types: [opened]
```

I dont think it blocks this PR as there are no web page changes.

**Review by eddiejaoude — APPROVED** [review]
Looks good 👍 - unable to test until merged, or we temporarily update the branch event.

**Comment by RemcoHalman** [comment]
> Thanks @RemcoHalman .
> 
> 
> 
> Interesting spot @BOLT04 , cypress should run when PRs are `opened` (we might need to add more events to this as PRs are updated and cypress should be re-run), this is the config we use
> 
> 
> 
> ```yaml
> 
> on:
> 
>   push:
> 
>   pull_request:
> 
>     types: [opened]
> 
> ```
> 
> 
> 
> I dont think it blocks this PR as there are no web page changes.

@eddiejaoude @BOLT04 A quick google got me me all the available types:
[docs link](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request)

I think adding **edited** should do the trick

**Comment by eddiejaoude** [comment]
thanks @RemcoHalman 👍 

This can be added in a new PR. Could you copy your example and link and create a new Issue

**Comment by eddiejaoude** [comment]
Oh you did already create an issue and PR - ignore my message above, I am still catching up on notifications 🤓

**Pull request merged**
